x-pack/plugins/endpoint/public/applications/endpoint/view/alerts/details/overview/index.tsx

```diff
@@ -4,91 +4,95 @@
  */
 import React, { memo, useMemo } from 'react';
+import styled from 'styled-components';
 import { i18n } from '@kbn/i18n';
 import { FormattedMessage } from '@kbn/i18n/react';
 import { EuiSpacer, EuiTitle, EuiText, EuiHealth, EuiTabbedContent } from '@elastic/eui';
 import { useAlertListSelector } from '../../hooks/use_alerts_selector';
 import * as selectors from '../../../../store/alerts/selectors';
+import { ResolverEvent } from '../../../../../../../common/types';
 import { MetadataPanel } from './metadata_panel';
 import { FormattedDate } from '../../formatted_date';
 import { AlertDetailResolver } from '../../resolver';
 
-export const AlertDetailsOverview = memo(() => {
-  const alertDetailsData = useAlertListSelector(selectors.selectedAlertDetailsData);
-  if (alertDetailsData === undefined) {
-    return null;
-  }
-  const selectedAlertIsLegacyEndpointEvent = useAlertListSelector(
-    selectors.selectedAlertIsLegacyEndpointEvent
-  );
+export const AlertDetailsOverview = styled(
+  memo(({ className }: { className?: string }) => {
+    const alertDetailsData = useAlertListSelector(selectors.selectedAlertDetailsData);
+    if (alertDetailsData === undefined) {
+      return null;
+    }
 
-  const tabs = useMemo(() => {
-    return [
-      {
-        id: 'overviewMetadata',
-        name: i18n.translate(
-          'xpack.endpoint.application.endpoint.alertDetails.overview.tabs.overview',
-          {
-            defaultMessage: 'Overview',
-          }
-        ),
-        content: (
-          <>
-            <EuiSpacer />
-            <MetadataPanel />
-          </>
-        ),
-      },
-      {
-        id: 'overviewResolver',
-        name: i18n.translate(
-          'xpack.endpoint.application.endpoint.alertDetails.overview.tabs.resolver',
-          {
-            defaultMessage: 'Resolver',
-          }
-        ),
-        content: (
-          <>
-            <EuiSpacer />
-            {selectedAlertIsLegacyEndpointEvent && <AlertDetailResolver />}
-          </>
-        ),
-      },
-    ];
-  }, [selectedAlertIsLegacyEndpointEvent]);
+    const tabs = useMemo(() => {
+      return [
+        {
+          id: 'overviewMetadata',
+          name: i18n.translate(
+            'xpack.endpoint.application.endpoint.alertDetails.overview.tabs.overview',
+            {
+              defaultMessage: 'Overview',
+            }
+          ),
+          content: (
+            <>
+              <EuiSpacer />
+              <MetadataPanel />
+            </>
+          ),
+        },
+        {
+          id: 'overviewResolver',
+          name: i18n.translate(
+            'xpack.endpoint.application.endpoint.alertDetails.overview.tabs.resolver',
+            {
+              defaultMessage: 'Resolver',
+            }
+          ),
+          content: (
+            <>
+              <EuiSpacer />
+              <AlertDetailResolver selectedEvent={(alertDetailsData as unknown) as ResolverEvent} />
```

Comment:
Why is this unsafe cast needed? can we work out a way to remove it?

Reply:
imo we should get rid of the AlertData type, as it's really just a variation of an EndpointEvent

Reply:
@oatkiller @kqualters-elastic @marshallmain `ResolverEvent` is defined as:
```javascript
ResolverEvent = EndpointEvent | LegacyEndpointEvent;
```
`AlertData` is defined as:
```javascript
export type AlertData = AlertEvent & AlertMetadata;
```
We have `malware_event`, `network`, `process`, and `registry` ECS schemas defined here: https://github.com/elastic/endpoint-app-team/tree/master/schemas/v1

Does it make sense to model our types after these schemas? If so, I think our `AlertEvent` types should follow the `malware_event` ECS and our `EndpointEvent` type should probably be a union of a new `ProcessEvent` type and our existing `AlertEvent`.

Reply:
yes agreed, wouldn't hurt to support `LegacyEndpointEvent` as well